src/components/CytoscapeGraph/ContextMenu/NodeContextMenu.tsx

```diff
@@ -49,17 +49,16 @@ const graphContextMenuItemLinkStyle = style({
 });
 
 type Props = NodeContextMenuProps & ReduxProps;
+type LinkParams = { namespace: string; name: string; type: string };
 
 export class NodeContextMenu extends React.PureComponent<Props> {
-  static derivedValuesFromProps(node: DecoratedGraphNodeData) {
+  static derivedValuesFromProps(node: DecoratedGraphNodeData): LinkParams | undefined {
     const namespace: string = node.namespace;
-    let name: string | undefined = '';
-    let type = '';
+    let name: string | undefined = undefined;
+    let type: string | undefined = undefined;
     switch (node.nodeType) {
-      case 'aggregate':
-        name = node.aggregateValue!;
-        break;
```

Comment:
Yes, if I'm correct in doesn't change the behaviour because the aggregate node contextual actions were disabled at a different level. My rationale is to have a single place where the logic happens (single source of truth), so "derivedValuesFromProps" now either provides the data necessary to build links for a node, or no data if links are not possible.

Reply:
Actually I should perhaps move the `node.isInaccessible` here too, to be consistent

Reply:
I think it's OK to omit it as we do have a default: case that sets nothing, and since aggregate nodes have no options I think it's OK to just fall through.  And the testing was fine.